### DISCUSSION TITLE
using the ip adress in ltsp.conf

### DISCUSSION THREAD

**Discussion by oosoG** [ltsp/ltsp#461] [Q&A]
Hi all,
During init of the client I need to write the ip adress + hostname + fully qualified hostname to /etc/hosts for some services. 
How can I do that?
`%{IP}` seems to work only in `HOSTNAME=` and it even is not the ip address.
It tried this (in ltsp.conf) in a `POST_INIT_CHANGE_HOSTS=` section but it seems that the ip isn't set yet

```
[change_hosts]
IP=`ifconfig $ip |grep -o -e 'inet addr:[^ ]*' | grep -o -e '[^:]*$'`
echo -n "\n\nip is $IP\n\n"

# set the hostname (otherwise $(hostname) does not give the correct

echo "$HOSTNAME" > /etc/hostname

cat << 'EOF' > /etc/hosts
127.0.0.1       localhost
::1             localhost ip6-localhost ip6-loopback
ff02::1         ip6-allnodes
ff02::2         ip6-allrouters
EOF
echo "$IP $HOSTNAME.domain.tld $HOSTNAME" >> /etc/hosts

```

**Comment by alkisg** [comment]
Hi, it's $IP_ADDRESS, not $IP.

The %{IP} pseudovariable is a completely different thing, documented in man ltsp.conf → HOSTNAME.

**Comment by oosoG** [comment]
Thank you @alkisg 
I'm using $IP_ADDRESS now but It still doesn't work ($IP_ADDRESS seems empty).
There are also these messages in the journal:
![image](https://user-images.githubusercontent.com/83859156/117544577-6b451c00-b022-11eb-8a63-2bb7163afc4b.png)
Once booted the network seems OK though

**Comment by alkisg** [comment]
Use `POST_INIT_DEBUG=debug_shell`
There, type `set`. You'll see all the environment variables, like IP_ADDRESS and MAC_ADDRESS.

If `ip -o route get 192.168.67.1` produces an error while the clients boot, you need to investigate it.
What is the output of `ip a` and `ip r` while at the debug shell?
I only saw this warning when people were booting an image locally (root=/dev/sda1 ltsp.image=x) instead of netbooting it.
